modules/planning/tasks/deciders/st_bounds_decider/st_obstacles_processor.cc

```diff
@@ -42,109 +43,148 @@ using apollo::common::PathPoint;
 using apollo::common::Status;
 using apollo::common::math::Box2d;
 using apollo::common::math::Vec2d;
+using apollo::common::math::LineSegment2d;
 
 STObstaclesProcessor::STObstaclesProcessor(
     const double planning_distance, const double planning_time,
-    const PathData& path_data) {
-  CHECK_GT(planning_time, 0.0);
-  planning_time_ = planning_time;
-
-  CHECK_GT(planning_distance, 0.0);
-  planning_distance_ = planning_distance;
-
-  CHECK_GT(path_data_.discretized_path().size(), 1);
-  path_data_ = path_data;
-}
+    const PathData& path_data)
+    : planning_time_(planning_time),
+      planning_distance_(planning_distance),
+      path_data_(path_data),
+      vehicle_param_(common::VehicleConfigHelper::GetConfig().vehicle_param())
+      {}
 
 Status STObstaclesProcessor::MapObstaclesToSTBoundaries(
     PathDecision* const path_decision) {
   // Sanity checks.
   CHECK_NOTNULL(path_decision);
+  CHECK_GT(planning_time_, 0.0);
+  CHECK_GT(planning_distance_, 0.0);
+  CHECK_GT(path_data_.discretized_path().size(), 1);
 
   // Go through every obstacle.
-
   return Status::OK();
 }
 
 // TODO(jiacheng): implement this.
-std::pair<double, double> STObstaclesProcessor::GetRegularBoundaryFromObstacles(
-    double t) {
-  return 0.0;
+std::pair<double, double>
+STObstaclesProcessor::GetRegularBoundaryFromObstacles(double t) {
+  return {0.0, 0.0};
 }
 
 // TODO(jiacheng): implement this.
 std::pair<double, double>
 STObstaclesProcessor::GetFallbackBoundaryFromObstacles(double t) {
-  return 0.0;
+  return {0.0, 0.0};
 }
 
 ///////////////////////////////////////////////////////////////////////////////
 // Private helper functions.
 
 bool STObstaclesProcessor::ComputeObstacleSTBoundary(const Obstacle& obstacle) {
-  std::vector<STPoint> lower_points;
-  std::vector<STPoint> upper_points;
-  const auto& adc_path_points = path_data_.discretized_path();
-  const auto& obs_trajectory = obstacle.Trajectory();
-
-  if (obs_trajectory.trajectory_point_size() == 0) {
-    // Processing a static obstacle.
-    // Sanity checks.
-    if (!obstacle.IsStatic()) {
-      AWARN << "Non-static obstacle[" << obstacle.Id()
-            << "] has NO prediction trajectory."
-            << obstacle.Perception().ShortDebugString();
-    }
-  } else {
-    // Processing a dynamic obstacle.
-  }
+  // std::vector<STPoint> lower_points;
```

Comment:
Add TODO to state when we should re-enable or retire this piece of code.

Reply:
Done.